2-promises/solution.js

```diff
@@ -24,16 +24,41 @@ const id = yourRandomMethod() //third run
 7. log the resultant fullname, or the error, at the end
 */
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
+const firstnames = require('./firstnames');
+const lastnames = require('./lastnames');
 
-    // You generate your id value here
+function solution() {
+  // YOUR SOLUTION GOES HERE
+  const fullName = [];
+  // You generate your id value here
+  const idFullName = Math.floor(Math.random() * 101);
 
-    // You call the lastnames method with your id
+  //   console.log(id);
+  // You call the lastnames method with your id
+  const getLastName = new Promise((resolve) => {
+    resolve(lastnames(idFullName));
+  });
 
-    // You call the firstname method
+  // You call the firstname method
+  getLastName
+    .then((lastname) => {
+      fullName.push(lastname);
+      return firstnames(lastname);
+    })
+    .catch((error) => console.log(error.message))
+    .then((name) => {
+      fullName.push(name);
```

Comment:
It looks like you're pushing the last name and then pushing the full name, is that what you wanted?

Reply:
In this case i checked the response and i saw that first is returning the lastname and then the first name so i need to improve my variable names to be understandable.